packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -153,4 +175,48 @@ export class AbapServiceProvider extends ServiceProvider implements AbapServiceP
         }
         return this.services[LayeredRepositoryService.PATH] as LayeredRepositoryService;
     }
+
+    /**
+     *
+     * @param packageName Package name for deployment
+     * @param appName Fiori project name for deployment. A new project that has not been deployed before is also allowed
+     * @returns
+     */
+    public async getTransportRequests(packageName: string, appName: string): Promise<string[]> {
+        const serviceSchema = await this.getAdtCatalogService().getServiceDefinition(
+            AdtServiceConfigs[AdtServiceName.TransportChecks]
+        );
+        // Service not available on target ABAP backend version, return empty setting config
+        if (!serviceSchema) {
+            return [];
+        }
+        const urlPath = serviceSchema.href;
```

Comment:
How about the case when `serviceSchema.href` is empty or `undefined`?

Reply:
We can assume href always presents if serviceSchema is not empty.

Reply:
Can we please code defensively and handle this?

Reply:
Thanks @slin-sap. Looks good. One of the last things to do: can we please log an error/warning when we're returning nothing? This would be a case when there's a bug/misconfiguration in the backend. Logging would expedite trouble-shooting